### ISSUE TITLE
FeatureRequest: support Arduino Nano Every (ATmega4809)

### ISSUE THREAD

**Issue by baumi51** [bxparks/AUnit#56]
My tests (and example tests) compile correct for Arduino UNO (ATmega328P) but NOT for Arduio Nano Every.

**Comment by bxparks** [comment]
Just to let you know that I am not completely ignoring this. I spent some time today scoping the work involved to add support for the megaAVR, and it does *not* seem easy.

The [Arduino megaAVR Core](https://github.com/arduino/ArduinoCore-megaavr/) significantly changed the programming framework in many backwards compatible breaking ways (e.g. https://github.com/arduino/ArduinoCore-API/issues/25). 

I looked at whether it would be possible to use the [MegaCoreX](https://github.com/MCUdude/MegaCoreX) instead. It was said that the "ATmega4809" board should work with the Nano Every. Unfortunately, the MegaCoreX core uses the [same F() hack](https://github.com/arduino/ArduinoCore-megaavr/issues/62) that is used in the Arduino megaAVR Core, which breaks AUnit in non-backwards compatible ways.

I'll poke at this intermittently as time allows, but it is possible that I will decide to make the megaAVR (e.g. the Nano Every) officially **unsupported**.
